src/poliastro/plotting.py

```diff
@@ -57,29 +55,36 @@ def __init__(self, ax=None, num_points=100):
         self.num_points = num_points
         self._frame = None
         self._attractor_radius = None
+        self.orbits = list(tuple())  # type: List[Tuple[poliastro.twobody.orbit.Orbit], str]
 
     def set_frame(self, p_vec, q_vec, w_vec):
-        """Sets perifocal frame if not existing.
+        """Sets perifocal frame.
 
         Raises
         ------
         ValueError
             If the vectors are not a set of mutually orthogonal unit vectors.
-        NotImplementedError
-            If the frame was already set up.
 
         """
-        if not self._frame:
-            if not np.allclose([norm(v) for v in (p_vec, q_vec, w_vec)], 1):
-                raise ValueError("Vectors must be unit.")
-            if not np.allclose([p_vec.dot(q_vec),
-                                q_vec.dot(w_vec),
-                                w_vec.dot(p_vec)], 0):
-                raise ValueError("Vectors must be mutually orthogonal.")
-            else:
-                self._frame = p_vec, q_vec, w_vec
+        old_frame = False
+        if self._frame is not None:
+            old_frame = True
+
+        if not np.allclose([norm(v) for v in (p_vec, q_vec, w_vec)], 1):
+            raise ValueError("Vectors must be unit.")
+        if not np.allclose([p_vec.dot(q_vec),
+                            q_vec.dot(w_vec),
+                            w_vec.dot(p_vec)], 0):
+            raise ValueError("Vectors must be mutually orthogonal.")
         else:
-            raise NotImplementedError
+            self._frame = p_vec, q_vec, w_vec
+
+        if old_frame:
```

Comment:
The complexity in this method is too high. Suggestions:

* Move `for artist ... artist.remove()` to a `_clear()` method
* Move `if old_frame ... self.plot(orbit, label)` to a `_redraw()`, `_reproject()` or `_replot()` method
* Try to get rid of the `old_frame` variable completely, for instance instead of `if old_frame` do `if self.orbits`, which will check if there are orbits to replot

Reply:
The main problem here is how the figure size is set. Now, figure size depend on whether there is a legend or not. Otherwise, `self.ax.clear()` could be used...
I'm trying to find a better solution, comments are welcome.